### ISSUE TITLE
symbols search works like full text search

### ISSUE THREAD

**Issue by vladak** [OpenGrok/OpenGrok#1376]
Symbol search works like full text search, i.e. it shows files/hits even for files that contain the symbol name in the comments. This means symbols search is broken.

This is observed in 0.13-rc3 with Universal ctags:

```
Universal Ctags 0.0.0(9c4c2de), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Jul 13 2016, 18:04:55
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +option-directory, +coproc, +xpath, +json
```

**Comment by vladak** [comment]
to reproduce this perform symbol search for e.g. `invalidateRepositories` in OpenGrok code.

**Comment by tarzanek** [comment]
can repro with exuberant ctags too, so this is a project bug

**Comment by tarzanek** [comment]
seems like my regression, this code:
 https://github.com/OpenGrok/OpenGrok/blob/master/src/org/opensolaris/opengrok/analysis/plain/AbstractSourceCodeAnalyzer.java#L75

doesn't kick in, and hence this code:
https://github.com/OpenGrok/OpenGrok/blob/master/src/org/opensolaris/opengrok/analysis/FileAnalyzer.java#L192
takes over and does a plain analysis, and not the one which ignores all comments, etc.

**Comment by tarzanek** [comment]
seems the inheritance doesn't kick in and the components are always taken from FileAnalyzer no matter what ...

**Comment by tarzanek** [comment]
I will need to add some more tests, than just fixing the 2 ones we had

**Comment by tarzanek** [comment]
fwiw  b1d8975 just fixes this for unit tests, indexer is still behaving wrongly, I need to figure out why lucene still takes FileAnalyzer as base, when we pass it on a different class.
This behaviour changed just recently by changes from @rmuir so I need to figure out how to  go around it ...

**Comment by tarzanek** [comment]
something between lucene 4.7.1 and 6.x made our AbstractSourceCodeAnalyzer to not use its overriden createComponents and FileAnalyzer is used always to create components

**Comment by tarzanek** [comment]
seems LUCENE-5634 is the cause

**Comment by tarzanek** [comment]
@mikemccand is there a way to get old behaviour of chain before LUCENE-5611 ?
So I can get OpenGroks overriden analyzers to work again ? (actually it smells a bit like a bug, but not sure, so better asking ... )

**Comment by tarzanek** [comment]
I think I see what is the cause now : https://issues.apache.org/jira/browse/LUCENE-6212
and I broke this in 6602c01097c66d242046fb0490e0a5dcc1ca36c5

**Comment by tarzanek** [comment]
fix is to use setTokenStream, resp. initialize the ref field with respective Tokenizer instead of relaying on components
we do the same with respective full fields

**Comment by tarzanek** [comment]
fix is in pull req #1429
